### PR TITLE
IAE-18889 Directory Returning Duplicate Results 

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.ts [hunk 1]
```diff
@@ -92,15 +93,16 @@ export class AutocompleteCache {
...
-    if (arr1.length !== arr2.length) {
-      return false;
-    }
-    for (let i = arr1.length; i--; ) {
-      if (arr1[i] !== arr2[i]) {
-        return false;
-      }
-    }
-    return true;
+    // if (arr1.length !== arr2.length) {
```
Comment (davereed) [inline src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.ts]: Remove commented code

File: src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.spec.ts [hunk 2]
```diff
@@ -118,3 +118,19 @@ describe('Sam Autocomplete Cache Class', () => {
     expect(cache.lastAdded).toEqual(moreValues);
   });
 });
+
+describe('Autocomplete Cache arraysEqual method', () => {
+  const initialArray = [{ key: 1 }, { key: 2 }, { key: 3 }, { key: 4 }];
```
Comment (davereed) [inline src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.spec.ts]: Add some more tests to cover more than just array:

https://github.com/lodash/lodash/blob/d5ef31929a1262abbc75b8dadc0b6ae6e9558b5f/test/isEqual.js

File: package.json [hunk 3]
```diff
@@ -46,7 +46,9 @@
     "ie-shim": "0.1.0",
     "rxjs": "^6.2.2",
     "zone.js": "0.8.14",
-    "moment": "2.19.3"
+    "moment": "2.19.3",
+    "@types/lodash": "4.14.123",
+    "lodash": "4.17.11"
```
Comment (davereed) [inline package.json]: Make sure artifactory has these versions

File: src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.ts (hunk 1 above)
Comment (yerramshilpa) [reply src/ui-kit/form-controls/autocomplete-multiselect/autocomplete-cache.ts]: Updated..

**Pull request merged**